src/App.js

```diff
@@ -70,35 +73,47 @@ class App {
 				// Success handler.
 				() => {
 					// Insert modified HTML.
-					$( '.mw-parser-output' ).html( self.api.getReplacementHtml() );
-					// Highlight when hover a user's contributions.
-					// @TODO This is just testing code and should be
-					// replaced by the proper behaviour.
-					$( '.mw-parser-output .editor-token' )
-						.on( 'mouseenter', function () {
-							const ids = self.api.getIdsFromElement( this );
-							// Activate all this user's contribution spans.
-							$( '.token-editor-' + ids.editorId ).addClass( 'active' );
-							// Information for popup.
-							// eslint-disable-next-line no-console
-							console.log( self.api.getTokenInfo( ids.tokenId ) );
-						} )
-						.on( 'mouseleave', function () {
-							// Deactivate all spans.
-							$( '.mw-parser-output .editor-token' ).removeClass( 'active' );
-						} );
-					self.widget.setState( 'ready' );
+					$( '.mw-parser-output' ).html( this.api.getReplacementHtml() );
+					this.attachContentListeners();
+					this.widget.setState( 'ready' );
 				},
 				// Error handler.
 				errorCode => {
-					self.widget.setState( 'err' );
-					self.widget.setErrorMessage( errorCode );
+					this.widget.setState( 'err' );
+					this.widget.setErrorMessage( errorCode );
 				}
 			);
 	}
 
 	/**
-	 * Repond to the close event that the widget emits.
+	 * Add listener to highlight attribution and show the RevisionPopupWidget.
+	 */
+	attachContentListeners() {
+		$( '.mw-parser-output .editor-token' )
+			.on( 'mouseenter', e => {
+				const ids = this.api.getIdsFromElement( e.target );
+				// Activate all this user's contribution spans.
+				$( '.token-editor-' + ids.editorId ).addClass( 'active' );
+			} )
+			.on( 'mouseleave', () => {
+				// Deactivate all spans.
+				$( '.mw-parser-output .editor-token' ).removeClass( 'active' );
+			} );
+
+		// Create RevisionPopupWidget instance.
```

Comment:
Should this be inside the "attachContentListeners" ? I think the creation of the revision popup should be either in the constructor (saved as class var) or in start(), and you can call it here with `this.revisionPopup` (or even "just" `this.popup` since it's probably the only popup in the system (welcome popup is a separate entry/operation code anyways)

In any case, the creation of the popup is unrelated to attaching listeners really, and having it be a callable class variable is more architecturally consistent and can actually be useful later.

Example: If we ever have the need to hide the popup programmatically (like, we might need to listen to VE's activation hook, for example) then having it as a class variable is useful.

Reply:
Sounds good to me! My thinking was we didn't want to create the popup until WWT actually finished loading, but I can at least move it to a separate method that doesn't have to do with adding listeners.